api/search.go

```diff
@@ -178,41 +159,157 @@ func CreateRequests(w http.ResponseWriter, req *http.Request, validator QueryPar
 		http.Error(w, "Invalid fromDate parameter", http.StatusBadRequest)
 		return "", nil, nil
 	}
-
 	toDateParam := paramGet(params, "toDate", "")
 	toDate, err := validator.Validate(ctx, "date", toDateParam)
 	if err != nil {
 		log.Warn(ctx, err.Error(), log.Data{"param": "toDateParam", "value": toDateParam})
 		http.Error(w, "Invalid toDate parameter", http.StatusBadRequest)
 		return "", nil, nil
 	}
-
 	if fromAfterTo(fromDate.(query.Date), toDate.(query.Date)) {
 		log.Warn(ctx, "fromDate after toDate", log.Data{"fromDate": fromDateParam, "toDate": toDateParam})
 		http.Error(w, "invalid dates - 'from' after 'to'", http.StatusBadRequest)
 		return "", nil, nil
 	}
 
-	// create SearchRequest with all the compulsory values
+	datasetIDs, datasetErr := processDatasetIDs(ctx, params)
+	if datasetErr != "" {
+		http.Error(w, datasetErr, http.StatusBadRequest)
+		return "", nil, nil
+	}
+
+	reqSearch := createSearchRequest(sanitisedQuery, offset, limit, contentTypes, fromDate.(query.Date), toDate.(query.Date), topics, sort, highlight, datasetIDs, nlpCriteria, params)
+	reqCount := createCountRequest(sanitisedQuery)
+
+	if debug {
+		log.Info(ctx, "[DEBUG]", log.Data{"search_request": reqSearch})
+	}
```

Comment:
I'm not particularly keen on propagating this pattern. We should be taking this from config really.